pandas/core/style.py

```diff
@@ -0,0 +1,777 @@
+            head.append(row_es)
+
+        body = []
+        for r, idx in enumerate(self.data.index):
+            cs = [ROW_HEADING_CLASS, "level%s" % c, "row%s" % r]
+            cs.extend(cell_context.get(
+                "row_headings", {}).get(r, {}).get(c, []))
+            row_es = [{"type": "th",
+                       "value": rlabels[r][c],
+                       "class": " ".join(cs)}
+                      for c in range(len(rlabels[r]))]
+
+            for c, col in enumerate(self.data.columns):
+                cs = [DATA_CLASS, "row%s" % r, "col%s" % c]
+                cs.extend(cell_context.get("data", {}).get(r, {}).get(c, []))
+                row_es.append({"type": "td", "value": self.data.iloc[r][c],
+                               "class": " ".join(cs), "id": "_".join(cs[1:])})
+                props = []
+                for x in ctx[r, c]:
+                    # have to handle empty styles like ['']
+                    if x.count(":"):
+                        props.append(x.split(":"))
+                    else:
+                        props.append(['', ''])
+                cellstyle.append(
+                    {'props': props,
+                     'selector': "row%s_col%s" % (r, c)}
+                )
+            body.append(row_es)
+
+        # uuid required to isolate table styling from others
+        # in same notebook in ipnb
+        return dict(head=head, cellstyle=cellstyle, body=body, uuid=uuid,
+                    precision=self.precision, table_styles=table_styles,
+                    caption=caption)
+
+    def render(self, table_styles=None, caption=None, uuid=None,
+               precision=None, trim=True):
+        """
+        Render the built up styles to HTML
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        table_styles: list or None
+            defaults to ``self.table_styles``
+        caption: str or None
+            defaults to ``self.caption``
+        uuid: str or None
+            defaults to random uuid
+        precision: int or None
+            defaults to self.precision
+        trim: bool
+            Small optimization to not create classes for cells that
+            have no styles applied to them.
+
+        Returns
+        -------
+        rendered: str
+            the rendered HTML
+
+        Notes
+        -----
+        ``Styler`` objects have defined the ``_repr_html_`` method
+        which automatically calls ``self.render()`` when it's the
+        last item in a Notebook cell. When calling ``Styler.render()``
+        directly, wrap the resul in ``IPython.display.HTML`` to view
+        the rendered HTML in the notebook.
+        """
+        self._compute()
+        table_styles = table_styles or self.table_styles
+        caption = caption or self.caption
+        uuid = uuid or self.uuid
+        d = self._translate(table_styles=table_styles, caption=caption,
+                            uuid=uuid)
+        if trim:
```

Comment:
Your probably right... this just trims from where the classes are referenced in the `style` block. I intended to go back and trim from ever creating the classes in the first place if they aren't used but didn't find a way to do that easily (in this case we would want a trim so that people can reference cells from an external style sheet)